content/docs/guides/tls-encryption.md

```diff
@@ -0,0 +1,109 @@
+
+```
+https://my-proxy.com/prometheus/metrics
+```
+
+Let's also say that you've generated the following using [OpenSSL](https://www.digitalocean.com/community/tutorials/openssl-essentials-working-with-ssl-certificates-private-keys-and-csrs) or an analogous tool:
+
+* an SSL certificate at `/root/certs/my-proxy.com/my-proxy.com.crt`
+* an SSL key at `/root/certs/my-proxy.com/my-proxy.com.key`
+
+You can generate a self-signed certificate and private key using this command:
+
+```bash
+mkdir -p /root/certs/my-proxy.com && cd /root/certs/my-proxy.com
+openssl req \
+  -x509 \
+  -newkey rsa:4096 \
+  -nodes \
+  -keyout my-proxy.com.key \
+  -out my-proxy.com.crt
+```
+
+Fill out the appropriate information at the prompts, and make sure to enter `my-proxy.com` at the `Common Name` prompt.
+
+## nginx configuration
+
+Below is an example [`nginx.conf`](https://www.nginx.com/resources/wiki/start/topics/examples/full/) configuration file. With this configuration, nginx will:
+
+* enforce TLS encryption using your provided certificate and key
+* proxy all connections to the `/prometheus` endpoint to a Prometheus server running on the same host (while removing the `/prometheus` from the URL)
+
+```conf
+events {}
+
+http {
+    server {
+        listen              443 ssl;
+        server_name         my-proxy.com;
+        ssl_certificate     /root/certs/my-proxy.com/my-proxy.com.crt;
+        ssl_certificate_key /root/certs/my-proxy.com/my-proxy.com.key;
+
+        location /prometheus {
+          proxy_pass http://localhost:9090/;
+        }
+    }
+}
+```
+
+Start nginx as root (since nginx will need to bind to port 443):
+
+```bash
+sudo nginx -c /usr/local/etc/nginx/nginx.conf
+```
+
+NOTE: This example uses `/usr/local/etc/nginx` as the location of the nginx configuration file, but this will vary based on the installation. Other [common nginx config directories](http://nginx.org/en/docs/beginners_guide.html) include `/usr/local/nginx/conf` and `/etc/nginx`.
+
+If nginx is already running, restart it with the new configuration:
+
+```bash
+sudo nginx -s reload -c /usr/local/etc/nginx/nginx.conf
+```
+
+NOTE: The nginx configuration above is incomplete and intended for example purposes. We recommend consulting the [official nginx documentation](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_certificate) when configuring nginx for TLS.
+
+## Prometheus configuration
+
+When running Prometheus behind the nginx proxy, you'll need to set the external URL to `http://my-proxy.com/prometheus` and the route prefix to `/`:
+
+```bash
+prometheus \
+  --config.file=/path/to/prometheus.yml \
+  --web.external-url=http://my-proxy.com/prometheus \
+  --web.route-prefix="/"
+```
+
+## Testing
+
+If you'd like to test out the nginx proxy locally using the `my-proxy.com` domain, you can add an entry to your `/etc/hosts` file that re-routes `my-proxy.com` to `localhost`:
+
+```
+127.0.0.1     my-proxy.com
```

Comment:
This is an actual website, we should probably use example.com for all this.

Reply:
Done